src/appleseed/renderer/kernel/lighting/bdpt/bdptlightingengine.cpp

```diff
@@ -78,6 +81,19 @@ namespace
             const ShadingPoint&     shading_point,
             ShadingComponents&      radiance) override      // output radiance, in W.sr^-1.m^-2
         {
+            PathVisitor light_path_visitor;
+            VolumeVisitor volume_visitor;
+
+            PathTracer<PathVisitor, VolumeVisitor, false> light_path_tracer(    // false = not adjoint
```

Comment:
For light tracing, you'll want to set `Adjoint` to `true`. You can find an extended treatment of adjoint tracing in Veach's thesis.